src/Makefile

```diff
@@ -16,7 +16,8 @@ SYMBOL_PREFIX	:=
 #
 # Locations of tools
 #
-HOST_CC		:= gcc
+CC		?= gcc
+HOST_CC		:= $(CC)
```

Comment:
This latest version of the change seems to affect only `HOST_CC`.  Since you still later have the line `CC := $(CROSS_COMPILE)gcc`, it looks as though this will always try to use a binary `gcc` for most of the build?

Reply:
See above comment.  The same concept works on FreeBSD and Linux by setting
CC=/usr/local/libexec/ccache/world/gcc12 or CC=/usr/lib/ccache/gcc.  I tested cross compiling on arm and arm64.

Reply:
With a test `Makefile` containing:
```makefile
$(warning CC=$(CC))
CC ?= gcc
$(warning CC=$(CC))
HOST_CC := $(CC)
$(warning HOST_CC=$(HOST_CC))
CC := $(CROSS_COMPILE)gcc
$(warning CC=$(CC))
all:
```
(i.e. matching this PR), I get the following:
```console
$ make
Makefile:1: CC=cc
Makefile:3: CC=cc
Makefile:5: HOST_CC=cc
Makefile:7: CC=gcc
make: Nothing to be done for 'all'.
```
but
```console
$ CC=some-other-compiler make
Makefile:1: CC=some-other-compiler
Makefile:3: CC=some-other-compiler
Makefile:5: HOST_CC=some-other-compiler
Makefile:7: CC=gcc
```
i.e. this definitely does **not** respect the value of CC set as an environment variable.

Reply:
> this definitely does not respect the value of CC set as an environment variable

```text
make CC=some-other-compiler
```
as in
```text
make -j 4 -C src NO_WERROR=1 V=1 \
 CROSS=arm-none-eabi- \
 CC=/usr/lib/ccache/arm-none-eabi-gcc \
   bin-arm32-efi/intel.efi \
   bin-arm32-efi/intel.usb \
   bin-arm32-efi/intel.iso
```
from above comment (fifty minutes ago).

Reply:
> ```
> make CC=some-other-compiler
> ```

That is irrelevant.  Passing variables as parameters on the `make` command line will always override whatever is set in the `Makefile`.  Even the current `Makefile` will "respect $CC" in that sense.  This PR is only relevant to a `CC` passed in as an environment variable.